README.md

```diff
@@ -1,61 +1,112 @@
-# jenkins-vault-plugin
+[![Build Status](https://jenkins.ci.cloudbees.com/buildStatus/icon?job=plugins/hashicorp-vault-plugin&build=20)](https://jenkins.ci.cloudbees.com/job/plugins/hashicorp-vault-plugin/20/)
+# Jenkins Vault Plugin
 
-This plugin adds a build wrapper to set environment variables from a HashiCorp [Vault](https://www.vaultproject.io/) secret.
+This plugin adds a build wrapper to set environment variables from a HashiCorp [Vault](https://www.vaultproject.io/) secret. Secrets are generally masked in the build log, so you can't accidentally print them.
 
-## Caveats
+# Vault Authentication Backends
+This plugin only allows authenticating against Vault using the AppRole authentication backend. Hashicorp recommends using AppRole for Servers / automated workflows (like Jenkins) and using Tokens (default mechanism, Github Token, ...) for every developer's machine.
```

Comment:
I mean line 7 where it states that the plugin only allows authenticating with approle.